pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -363,12 +363,69 @@ func TestPriorityQueue_MoveAllToActiveOrBackoffQueue(t *testing.T) {
 	q.Add(&medPriorityPod)
 	q.AddUnschedulableIfNotPresent(q.newQueuedPodInfo(&unschedulablePod), q.SchedulingCycle())
 	q.AddUnschedulableIfNotPresent(q.newQueuedPodInfo(&highPriorityPod), q.SchedulingCycle())
+	// sleep 1 second so that pods is guaranteed to be added into unschedulableQ first and then
+	// the move request is happened later, otherwise there is potential a test flake here, as the
+	// move request has a chance to be finished before the pods get into the unschedulableQ.
+	// In that case, the pods will be moved to backoffQ even `QueuedPodInfo.Attempts` is zero.
+	time.Sleep(1000 * time.Millisecond)
 	q.MoveAllToActiveOrBackoffQueue("test")
+	if q.activeQ.Len() != 3 {
+		t.Error("Expected 3 item to be in activeQ")
+	}
+	if q.podBackoffQ.Len() != 0 {
+		t.Error("Expected 0 items to be in podBackoffQ")
+	}
+}
+
+// TestPriorityQueue_RaceConditionOnScheduling mimic a race condition could be found with preemption.
+// In this particular case, the high priority pod is stuck in the backoffQ and if new pod is coming,
+// regardless it's priority, the new pod will be scheduled first.
+// The preemption could happen again but there is a chance the pod will never got scheduled as long as
+// there is new pod coming for scheduling since the high priority pod could be moved from backoffQ
+// to activeQ only when the time for backoff expiration is up.
+// see https://github.com/kubernetes/kubernetes/issues/93505 for the details.
```

Comment:
I'm not sure how is this related. Could you document your findings in the issue?

Reply:
updated in that issue, pls kindly check it at your convenient time.